rpcserverhandlers_test.go

```diff
@@ -652,6 +652,24 @@ type rpcTest struct {
 	errCode         dcrjson.RPCErrorCode
 }
 
+// Provide default configurations that can be overridden by tests as needed.
+var defaultCfg = &config{}
+var defaultChainParams = chaincfg.MainNetParams()
+var defaultRPCChain = &testRPCChain{
+	bestSnapshot: &blockchain.BestState{
+		Height: 463073,
+		Bits:   404696953,
+		Hash: *mustParseHash("00000000000000001e6ec1501c858506de1de4703d1be8bab4061" +
+			"126e8f61480"),
+		PrevHash: *mustParseHash("00000000000000001a1ec2becd0dd90bfbd0c65f42fdaf608" +
+			"dd9ceac2a3aee1d"),
+	},
```

Comment:
Added a default `bestSnapshot` here. I'll add some other defaults in a separate PR as well but this unblocks the current issue.

Reply:
Right.  In general, I think it would make the most to sense to ensure there is always a sane default config that satisfies all interfaces and then the tests just override the things they care about.

I typically handle this via the use of a test harness helper of some sort.  You can take a look at `mempool` for one such example if you like, but essentially the idea is that each test does something like:

```Go
// Start with a fully functional mock implementation and override test values...
mockConfig := defaultMockConfig()
if test.whatever != nil {
	mockConfig.whatever = test.whatever
}
// ...
testServer, err := newRPCServer(mockConfig)
if err != nil {
	t.Fatalf(...)
}
```

That way it actually fully creates the RPC server as happens in the real server code.